### PR TITLE
admin bulk enroll api for the batch in a program

### PR DESCRIPTION
**Pull request by shankaragoudab**

### CODE REVIEW

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala [hunk 1]
```diff
@@ -692,6 +694,63 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
...
+                response.put(userId, status)
+            } catch {
+                case e: ProjectCommonException =>
+                    if (ResponseCode.userAlreadyEnrolledCourse.getErrorMessage.equals(e.getMessage)) {
+                        map.put(JsonKey.STATUS, JsonKey.FAILED)
+                        map.put(JsonKey.ERRORMSG, ResponseCode.userAlreadyEnrolledCourse.getErrorMessage)
+                        response.put(userId, map)
+                    } else {
+                        map.put(JsonKey.STATUS, JsonKey.FAILED)
+                        map.put(JsonKey.ERRORMSG, "user not enrolled")
```
Comment (karthik-tarento) [inline course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: errMsg -- should explain what is the actual error... get the information from exception and use the same.

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala [hunk 2]
```diff
@@ -692,6 +694,63 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
...
+                        response.put(userId, status)
+                    }
+                case e: Exception =>
+                    map.put(JsonKey.STATUS, JsonKey.FAILED)
+                    map.put(JsonKey.ERRORMSG, "user not enrolled")
```
Comment (karthik-tarento) [inline course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: errMsg -- should explain what is the actual error... get the information from exception and use the same.

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/BaseRequestValidator.java [hunk 3]
```diff
@@ -499,4 +501,12 @@ public void validateRequestedBy(String requestedBy) {
               ResponseCode.UNAUTHORIZED.getResponseCode());
     }
   }
+  public void validateParamCollection(List<String> value, ResponseCode error,String errorMsgArgument) {
+    if (CollectionUtils.isEmpty (value)||null==(value)) {
```
Comment (karthik-tarento) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/BaseRequestValidator.java]: CollectionUtils.isEmpty -- will check the given object is null or not... no need of additional null value... 
But you need to loop through the collection and make sure empty string is given as input

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala (hunk 1 above)
Comment (shankaragoudab) [reply course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: resolved

File: course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala (hunk 2 above)
Comment (shankaragoudab) [reply course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala]: resolved

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/BaseRequestValidator.java (hunk 3 above)
Comment (shankaragoudab) [reply course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/BaseRequestValidator.java]: 1. removed this part in the code   ||null==(value) 
2. collection is iterated and throwing error for empty string,
 in CourseEnrollmentRequestValidator line 101 to 107
 
    **if (userIdList != null&& userIdList.size()<30) {
      for (String userId : userIdList) {
        validateParam(
                userId,ResponseCode.invalidParameterValue,
                JsonKey.USER_ID);**

**Pull request merged**